src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionClosePaymentRetryQueueConsumer.kt

```diff
@@ -125,8 +126,14 @@ class TransactionClosePaymentRetryQueueConsumer(
                     trxWithAuthorizationCompleted ->
                     when (trxWithAuthorizationCompleted.transactionAuthorizationCompletedData
                       .authorizationResultDto) {
-                      AuthorizationResultDto.OK -> ClosePaymentRequestV2Dto.OutcomeEnum.OK
-                      AuthorizationResultDto.KO -> ClosePaymentRequestV2Dto.OutcomeEnum.KO
+                      AuthorizationResultDto.OK ->
+                        Either.right(
+                          Tuple.of(
+                            ClosePaymentRequestV2Dto.OutcomeEnum.OK,
+                            trxWithAuthorizationCompleted.transactionAuthorizationCompletedData
+                              .authorizationCode))
+                      AuthorizationResultDto.KO ->
+                        Either.left(ClosePaymentRequestV2Dto.OutcomeEnum.KO)
```

Comment:
Given that `Either.right` corresponds to OK and `Either.left` corresponds to KO I would use an `Optional<String>` with the authorization code instead